logstash-core/src/main/java/org/logstash/ackedqueue/AckedReadBatch.java

```diff
@@ -0,0 +1,84 @@
+package org.logstash.ackedqueue;
+
+import org.jruby.RubyArray;
+import org.jruby.RubyHash;
+import org.jruby.runtime.ThreadContext;
+import org.jruby.runtime.builtin.IRubyObject;
+import org.logstash.ackedqueue.ext.JRubyAckedQueueExt;
+import org.logstash.execution.MemoryReadBatch;
+import org.logstash.execution.QueueBatch;
+import org.logstash.ext.JrubyEventExtLibrary;
+
+import java.io.IOException;
+import java.util.Collection;
+
+import static org.logstash.RubyUtil.RUBY;
+
+public final class AckedReadBatch implements QueueBatch {
+
+    private AckedBatch ackedBatch;
+
+    private RubyHash originals;
+
+    private RubyHash generated;
+
+    public static AckedReadBatch create(final JRubyAckedQueueExt queue, final int size,
+                                        final long timeout) {
+        return new AckedReadBatch(queue, size, timeout);
+    }
+
+    private AckedReadBatch(final JRubyAckedQueueExt queue, final int size, final long timeout) {
+        ThreadContext context = RUBY.getCurrentContext();
+        AckedBatch batch = null;
+        try {
+            batch = queue.readBatch(size, timeout);
+        } catch (IOException e) {
+            // what needs to be done here?
```

Comment:
@danhermann let's rethrow `new IllegalStateException(e)` here. We're dead if a read from disk throws ... let's just err out on a `RuntimeException` accordingly:)